### PR TITLE
Added K-NN APIs get_model, search_model, delete_model, train_model

### PR DESCRIPTION
**Pull request by saimedhi**
### Description

- Added K-NN APIs get_model, search_model, delete_model, train_model

- Tested the api with different path and query params [here](https://github.com/saimedhi/opensearch_api_testing/blob/main).

- k-NN doc https://opensearch.org/docs/latest/search-plugins/knn/api

- plugin code ref: https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/plugin/rest

### Issues Resolved
Related to https://github.com/opensearch-project/opensearch-api-specification/issues/168,
https://github.com/opensearch-project/opensearch-py/issues/300

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by nhtruong** [comment]
There's a Smithy Build error: https://github.com/opensearch-project/opensearch-api-specification/actions/runs/8011904023/job/21886085420?pr=192https://github.com/opensearch-project/opensearch-api-specification/actions/runs/8011904023/job/21886085420?pr=192

**Comment by saimedhi** [comment]
> There's a Smithy Build error: https://github.com/opensearch-project/opensearch-api-specification/actions/runs/8011904023/job/21886085420?pr=192https://github.com/opensearch-project/opensearch-api-specification/actions/runs/8011904023/job/21886085420?pr=192



> There's a Smithy Build error: https://github.com/opensearch-project/opensearch-api-specification/actions/runs/8011904023/job/21886085420?pr=192https://github.com/opensearch-project/opensearch-api-specification/actions/runs/8011904023/job/21886085420?pr=192

Sorry, Its passing now.

**Comment by saimedhi** [comment]
@Xtansia, @nhtruong Please take a look and let me know if any changes needed.

File: model/knn/search_model/operations.smithy [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+//  compatible open source license.
+
+$version: "2"
+namespace OpenSearch
+
+@externalDocumentation(
+    "API Reference": "https://opensearch.org/docs/latest/search-plugins/knn/api/#search-model"
+)
+
+@xOperationGroup("knn.search_model")
```
Comment (Xtansia) [inline model/knn/search_model/operations.smithy:14]: This should be pluralized as `knn.search_models`

**Review by Xtansia — CHANGES_REQUESTED** [review]

File: model/knn/search_model/operations.smithy (hunk 1 above)
Comment (saimedhi) [reply model/knn/search_model/operations.smithy]: @Xtansia, made the requested changes.

**Review by Xtansia — APPROVED** [review]

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
@saimedhi Try adding the same APIs to the OpenAPI feature branch so we can compare? https://github.com/opensearch-project/opensearch-api-specification/tree/feature/native_openapi

**Pull request merged**